src/plugins/workspace/server/plugin.ts

```diff
@@ -83,8 +88,35 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
     if (isPermissionControlEnabled) {
```

Comment:
Shall we move the permission control logic to a separate method like `setupPermission`? It seems over 30 lines.

Reply:
Sure, I have moved the related logic to a separate `setupPermission` method.